src/pages/Register.tsx

```diff
@@ -0,0 +1,283 @@
+    </LabeledCheckbox>
+  </div>
+);
+
+type InterestInTerraCheckboxProps = {
+  title: string;
+  interestInTerra: string;
+  setFunc: (interest: string) => void;
+};
+
+const InterestInTerraCheckbox = (props: InterestInTerraCheckboxProps) => (
+  <div style={{ marginTop: '.25rem' }}>
+    <LabeledCheckbox
+      checked={_.includes(props.title, props.interestInTerra)}
+      disabled={false}
+      onChange={(v: string) => {
+        const interestsList = _.isEmpty(props.interestInTerra) ? [] : _.split(',', props.interestInTerra);
+        const updatedInterestsList = v
+          ? _.concat(interestsList, [props.title])
+          : _.without([props.title], interestsList);
+        props.setFunc(_.join(',', updatedInterestsList));
+      }}
+    >
+      <span style={{ marginLeft: '0.5rem' }}>{props.title}</span>
+    </LabeledCheckbox>
+  </div>
+);
+
+const Register = () => {
+  const user: TerraUser = getTerraUser();
+  const userAttributes: SamUserAttributes = authStore.get().terraUserAttributes;
+  const [busy, setBusy] = useState(false);
+  const [givenName, setGivenName] = useState(user.givenName || '');
+  const [familyName, setFamilyName] = useState(user.familyName || '');
+  const [email, setEmail] = useState(user.email || '');
+  const [partOfOrganization, setPartOfOrganization] = useState(true);
+  const [institute, setInstitute] = useState(''); // keep this key as 'institute' to be backwards compatible with existing Thurloe KVs
+  const [title, setTitle] = useState('');
+  const [department, setDepartment] = useState('');
+  const [interestInTerra, setInterestInTerra] = useState('');
+  const [marketingConsent, setMarketingConsent] = useState(userAttributes.marketingConsent);
+
+  const register = async () => {
+    try {
+      setBusy(true);
+      const orgFields = partOfOrganization
+        ? {
+            institute,
+            department,
+            title,
+          }
+        : {};
+      await Ajax().User.profile.set({
+        firstName: givenName,
+        lastName: familyName,
+        contactEmail: email,
+        interestInTerra,
+        ...orgFields,
+      });
+      await Ajax().User.setUserAttributes({ marketingConsent });
+      authStore.update((state) => ({ ...state, registrationStatus: 'registeredWithoutTos' }));
+      await refreshTerraProfile();
+      Ajax().Metrics.captureEvent(Events.user.register);
+    } catch (error) {
+      reportError('Error registering', error);
+      setBusy(false);
+    }
+  };
+  const errors = validate(
+    { givenName, familyName, email, institute, title, department },
+    constraints(partOfOrganization)
+  );
+
+  return (
+    <div
+      role="main"
+      style={{
+        flexGrow: 1,
+        padding: '5rem',
+        backgroundImage: `url(${planet})`,
+        backgroundRepeat: 'no-repeat',
+        backgroundSize: '750px',
+        backgroundPosition: 'right 0px bottom -600px',
+      }}
+    >
+      {registrationLogo()}
```

Comment:
JSX definitely makes these render functions seem more out of place.

Reply:
perhaps converting it to a functional component is in order 🤔